src/tree/AppResourceTreeItem.ts

```diff
@@ -128,9 +128,10 @@ export class AppResourceTreeItem extends ResolvableTreeItemBase implements Group
         // do nothing as we only want to return parent dynamically
     }
 
-    public async refreshImpl(): Promise<void> {
+    public async refreshImpl(context: IActionContext): Promise<void> {
         this.mTime = Date.now();
         ext.tagFS.fireSoon({ type: FileChangeType.Changed, item: this });
+        await super.refreshImpl(context);
```

Comment:
Shouldn't refresh be dumping all of these nodes entirely?

Reply:
This refresh implementation is for a single app resource, which nodes are you referring to? The children are automatically cleared from cache on refresh by utils.

Reply:
The bug is that when we refresh, we aren't re-resolving, right? I think that we should entirely clear the cache (i.e. the `ResourceCache`), including the tree nodes, resolve results, etc., when we refresh. This issue would be fixed if we can trigger a refresh when these extensions install.

Reply:
It sounds like you want to refresh and re-resolve all resources when an extension is installed. I think installing the Functions extension should only refresh and re-resolve Function app resources instead of all app resources.

Reply:
That's definitely reasonable, but is it worth the extra work?

Reply:
I also believe refreshing a group should re-resolve all children. And refreshing an app resource should re-resolve it. The code I added also adds this, so the extra work is probably needed anyway.